### ISSUE TITLE
Property of type object without subproperties throws unhelpful error

### ISSUE THREAD

**Issue by iann0036** [aws-cloudformation/cloudformation-cli-go-plugin#104]
When a property is an object without any subproperties, like:

```
        "Tags": {
            "type": "object"
        },
```

The error thrown is unhelpful. Consider checking for this case and produce better error handling.

```$ cfn generate
resource/model.go:30:13: expected ';', found '<'
resource/model.go:34:3: expected '}', found 'EOF'
exit status 2
=== Caught downstream error ===
Command '['go', 'fmt', PosixPath('_..._/cmd/resource/model.go')]' returned non-zero exit status 1.
---
If debugging indicates this is a possible error with this program,
please report the issue to the team and include the log file 'rpdk.log'.
Issue tracker: https://github.com/aws-cloudformation/aws-cloudformation-rpdk/issues
```

**Comment by jeshan** [comment]
Ian, I suggest you use proper definitions for nested objects; it's cleaner and bypasses this issue.

e.g instead of having


```json
"properties": {
        "MyProperty": {
            "items": {
                "type": "object"
            },
            "type": "array"
        }
}
```
do this instead:
```json
"definitions": {
        "MyItem": {
            "type": "object",
            "properties": {
                "Prop1": {
                    "type": "string"
                },
                "Prop2": {
                    "type": "string"
                }
            }
        }
    }
}
```
```json
"properties": {
        "MyProperty": {
            "items": {
                "$ref": "#/definitions/MyItem"
            },
            "type": "array"
        }
}
```

I learned this here:
https://json-schema.org/learn/miscellaneous-examples.html

**Comment by stilvoid** [comment]
Fixed in #106

**Comment by brianterry** [comment]
I'm closing this issue because it was fixed in #106 .
